### PR TITLE
Only render add comment component for users who have comment permissions; add view-only user role; add test

### PR DESCRIPTION
**Pull request by acrellin**
Closes https://github.com/skyportal/skyportal/issues/207

### CODE REVIEW

File: skyportal/tests/frontend/test_sources.py [hunk 1]
```diff
@@ -84,3 +84,16 @@ def test_download_comment_attachment(driver, user, public_source):
...
+def test_view_only_user_cannot_comment(driver, view_only_user, public_source):
+    driver.get(f"/become_user/{view_only_user.id}")
+    driver.get(f"/source/{public_source.id}")
+    driver.wait_for_xpath(f'//div[text()="{public_source.id}"]')
+    try:
+        driver.wait_for_xpath('//input[@name="comment"]', timeout=0)
+    except:
+        pass
+    else:
+        raise Exception('Comment form is visible to view-only user, but should be hidden.')
```
Comment (stefanv) [inline skyportal/tests/frontend/test_sources.py]: Nicely done.

This seems to become a pattern, so we can also add a little shim: `driver.assert_missing_xpath(XPATH, error message)`?  Will leave that for another PR.

File: static/js/components/CommentList.jsx [hunk 2]
```diff
@@ -50,7 +51,10 @@ const CommentList = () => {
     <div className={styles.comments}>
       {items}
       <br />
-      <CommentEntry addComment={addComment} />
+      {
```
Comment (stefanv) [inline static/js/components/CommentList.jsx]: Curious about logic; would a user not sometimes have comment rights *only on certain sources*?

Comment (acrellin) [reply static/js/components/CommentList.jsx]: Hmm, good question... We don't currently have that concept, but it's something we should consider adding. Currently the endpoint handler just checks whether the user has the "Comment" ACL: `@permissions(['Comment'])` - I'll open an issue to look into adding per-source (or per-group) permissions

File: skyportal/tests/frontend/test_sources.py (hunk 1 above)
Comment (acrellin) [reply skyportal/tests/frontend/test_sources.py]: Good call - after this one gets merged, I'll add that to baselayer and update all the skyportal tests (including this one) accordingly.
